lib/compiler/passes/generate-bytecode.js

```diff
@@ -313,7 +320,8 @@ function generateBytecode(ast) {
     },
 
     named(node, context) {
-      const nameIndex = addConst(
+      const match = node.match | 0;
```

Comment:
why are you bitwise oring node.match with zero? what does this mean?

Reply:
Not really, no.

Reply:
It turns out this is a substitute for `(if node.?match === 0)`

This is bad code and should not land.  We are not creating a trivia contest.